### DISCUSSION TITLE
Can smoltcp be used as a virtual userspace networking stack?

### DISCUSSION THREAD

**Discussion by clarkmcc** [smoltcp-rs/smoltcp#978] [Q&A]
I'm not familiar with this project, but I noticed that it has nice abstractions at multiple networking layers so it got me thinking, could I use this to run user-space networking? Specifically, I'm looking for a Rust replacement for [Google's netstack](https://github.com/google/netstack) so that I can tunnel L2 over something else (like L4). See [this project](https://github.com/clarkmcc/remotenetstack) as an example.

**Comment by strobecat** [comment]
I'm trying to make one, for Python strobecat/pytun-pmd3
Though facing some problems now
